sorc/ncep_post.fd/UPP_PHYSICS.f

```diff
@@ -2638,6 +2641,709 @@ SUBROUTINE CALGRADPS(PS,PSX,PSY)
+         read(11,*) datacount, outputAxon
+      endif
+
+      if(trim(bbstring)=='#hidden1Synapse FullSynapse') then
+         read(11,*) datacount, hidden1Synapse
+      endif
+
+      if(trim(bbstring)=='#outputSynapse FullSynapse') then
+         read(11,*) datacount, outputSynapse
+      endif
+
+      if(trim(bbstring)=='#activeOutputProbe DataWriter') then
+         read(11,*) datacount
+         do j=1,3
+            read(11,*) activeOutputProbe(:,j)
+         enddo
+      endif
+
+      enddo
+
+      close(11)
+
+      if(activeOutputProbe(1,1)==1.) then
+      do j=1,3
+         activeOutputProbe(1,j)=8.999999761581421e-001
+         activeOutputProbe(2,j)=5.000000074505806e-002
+      enddo
+      endif
+
+! Run Network
+
+      do j=1,7
+         inputAxon(j) = inputFile(1,j) * f(j) + inputFile(2,j)
+      enddo
+
+      fgrid1=0.
+!$omp parallel do private(i,j)
+      do j=1,40
+         do i=1,7
+            fgrid1(j) = fgrid1(j) + hidden1Synapse(i,j) * inputAxon(i) 
+         enddo
+         fgrid1(j) = fgrid1(j) + hidden1Axon(j)
+         fgrid1(j) = (exp(fgrid1(j))-exp(-fgrid1(j)))/(exp(fgrid1(j))+exp(-fgrid1(j)))
+      enddo
+
+      fgrid2=0.
+      fgridsum=0.
+      do j=1,3
+         do i=1,40
+            fgrid2(j) = fgrid2(j) + outputSynapse(i,j) * fgrid1(i)
+         enddo
+         fgrid2(j) = fgrid2(j) + outputAxon(j)
+         fgrid2(j) = exp(fgrid2(j))
+         fgridsum = fgridsum + fgrid2(j)
+      enddo
+      do j=1,3
+         fgrid2(j) = fgrid2(j) / fgridsum
+!         fgrid2(j) = activeOutputProbe(1,j) * fgrid2(j) + activeOutputProbe(2,j)  
+      enddo
+
+      p1 = fgrid2(1)
+      p2 = fgrid2(2)
+      p3 = fgrid2(3)
+
+      END SUBROUTINE breadboard1_main
+!
+!-------------------------------------------------------------------------------------
+!
+      SUBROUTINE breadboard6_main(nswFileName,mf,f1,f2,f3,f4,f5,f6,p1,p2,p3)
+
+      implicit none
+
+      character*20 nswFileName
+      real mf, f1, f2, f3, f4, f5, f6
+      real p1, p2, p3
+
+      real f(7)
+
+      integer ieof
+      character*100 bbstring
+      integer datacount
+      real inputFile(2,7)
+      real inputAxon(7)
+      real hidden1Axon(7)
+      real hidden2Axon(4)
+      real outputAxon(3)
+      real hidden1Synapse(7,7)
+      real hidden2Synapse(7,4)
+      real outputSynapse(4,3)
+      real activeOutputProbe(2,3)
+
+      real fgrid1(7), fgrid2(4), fgrid3(3), fgridsum
+
+      integer i,j
+!
+      f(1) = mf
+      f(2) = f1
+      f(3) = f2
+      f(4) = f3
+      f(5) = f4
+      f(6) = f5
+      f(7) = f6
+!
+      inputFile(1,:)=1.
+      inputFile(2,:)=0.
+      inputAxon=0.
+      hidden1Axon=0.
+      hidden2Axon=0.
+      outputAxon=0.
+      hidden1Synapse=1.
+      hidden2Synapse=1.
+      outputSynapse=1.
+      activeOutputProbe(1,:)=1.
+      activeOutputProbe(2,:)=0.
+
+      open(11,file=nswFileName,status='unknown')
+
+      ieof = 0
+      do while (ieof == 0)
+
+      read(11,'(a)',iostat=ieof) bbstring
```

Comment:
@JesseMeng-NOAA I think the reading of Breadnboard* files don't have to be done at each grid point as it does not
depend on model inputs.  If you can redesign this module and read these Breadboard files once before main SLR i/j loop
and save them to local arrays to be used inside of SLR i/j loop,  I think you will save time.  I suspect the variable run times Wen saw may be attributed to the fact you're reading these files 3000x1500 times.  IO are time consuming and do present more variabilities.

Reply:
@JesseMeng-NOAA @WenMeng-NOAA 
Given the long and unstable run time with generating SLR, let's wait to merge this PR.

Reply:
Agree.